ydb/docs/en/core/integrations/import-jdbc.md

```diff
@@ -0,0 +1,256 @@
+        <!-- The particular tables can be included or excluded using the
+             "include-tables" and "exclude-tables" tags,
+             using literals or regular expressions as well. -->
+    </table-map>
+    <!-- Table reference may refer to the particular table in the particular schema,
+         and optionally specify the query to execute. The latter option allows to import
+         the virtual tables which do not actually exist on the data source. -->
+    <table-ref options="default">
+        <schema-name>ora$sys</schema-name>
+        <table-name>all_tables</table-name>
+        <!-- If the query text is defined, it is executed as shown. -->
+        <query-text>SELECT * FROM all_tables</query-text>
+        <!-- Primary key columns should be defined for the query.  -->
+        <key-column>OWNER</key-column>
+        <key-column>TABLE_NAME</key-column>
+    </table-ref>
+</ydb-importer>
+```
+
+## Import tool limitations {#limitations}
+
+Right now the tested data sources include the following:
+
+* [PostgreSQL](https://www.postgresql.org/),
+* [MySQL](https://www.mysql.com/),
+* [Oracle Database](https://www.oracle.com/database/),
+* [Microsoft SQL Server](https://www.microsoft.com/sql-server/),
+* [IBM Db2](https://www.ibm.com/products/db2),
+* [IBM Informix](https://www.ibm.com/products/informix).
+
+Other data sources will probably work, too, as the tool uses the generic JDBC APIs to retrieve data and metadata.
+
+Secondary indexes are not imported.
+
+Some data types and table structures are known to be unsupported:
+
+* the embedded tables for Oracle Database,
+* spatial data type for Microsoft SQL Server,
+* object data types for Informix.
+
+### Handling tables without the primary key {#nopk}
+
+Each YDB table must have a primary key. If a primary key (or at least a unique index) is defined on the source table, the tool creates the primary key for the target YDB table with the columns and order defined by the original primary key. When having multiple unique indexes defined on the source table, the tool prefers the index with the smallest number of columns.
+
+In case the source table has no primary key and no unique indexes, primary key columns can be configured using the import settings in the configuration file, as a sequence of `key-column` elements in the `table-ref` section (see below the example in the description of the configuration file format).
+
+When there is no primary key defined anywhere, the tool automatically adds the column `ydb_synth_key` to the target table, and uses it as the primary key. The values of the synthetic primary key are computed as "SHA-256" hash code over the values of all columns of the row, except the columns of the BLOB type.
+
+If the input table contains several rows with completely identical values, the destination table will have only one row per each set of duplicate input rows. This also means that there the tool cannot import the table in which all columns are of BLOB type.
```

Comment:
```suggestion
### Handling tables without the primary key {#nopk}

Each {{ ydb-short-name }} table must have a primary key. If a primary key (or at least a unique index) is defined on the source table, the tool creates the primary key for the target  {{ ydb-short-name }} table with the columns and order determined by the original primary key. When multiple unique indexes are defined on the source table, the tool prefers the index with the smallest number of columns.

In case the source table has no primary key and no unique indexes, primary key columns can be configured using the import settings in the configuration file as a sequence of `key-column` elements in the `table-ref` section (see below the example in the description of the configuration file format).

When no primary key is defined anywhere, the tool automatically adds the column `ydb_synth_key` to the target table and uses it as the primary key. The values of the synthetic primary key are computed as "SHA-256" hash code over the values of all row cells except the columns of the `BLOB` type.

If the input table contains several rows with completely identical values, the destination table will have only one row per each set of duplicate input rows. This also means that the tool cannot import the table in which all columns are of `BLOB` type.
```